LINBIT/Linstor.pm

```diff
@@ -255,20 +260,60 @@ sub set_vmid {
     return 1;
 }
 
+sub resource_uses_nvme {
+    my ( $self, $name ) = @_;
+
+    # Query resource definition to check layer list
+    my $ret = $self->{cli}->GET("/v1/resource-definitions/$name");
+    return 0 unless $ret->responseCode() eq '200';
+
+    my $rd;
+    eval { $rd = decode_json( $ret->responseContent() ); };
+    return 0 if $@;
+
+    # Check if NVME is in the layer list
+    my $layer_stack = $rd->{layer_data} || [];
+    foreach my $layer (@$layer_stack) {
+        return 1 if uc($layer->{type}) eq 'NVME';
+    }
```

Comment:
do we need to be stricter here? could people create - or worse have - strange setups where they run into problems with that? maybe we want to check where in they layer list NVME is? if it is the only layer? the bottom layer?

Reply:
after talking to the devs: mixing nvme with something else like DRBD is not supported in LINSTOR, so checking for 'NVME' should be good enough